src/test/java/persistence/study/ReflectionTest.java

```diff
@@ -0,0 +1,130 @@
+        Arrays.stream(fields).forEach(field -> {
+            logger.info("name: " + field.getName() + ", type: " + field.getType());
+        });
+    }
+
+    @Test
+    @DisplayName("Car 객체 생성자 정보 가지고 오기 ")
+    void showClass_2() {
+        Class<Car> carClass = Car.class;
+        Constructor<?>[] constructors = carClass.getDeclaredConstructors();
+        Arrays.stream(constructors).forEach(constructor -> {
+            logger.info("Constructor: " + constructor.getName() + ", Parameter Types " + Arrays.toString(constructor.getParameterTypes()));
+        });
+    }
+
+    @Test
+    @DisplayName("Car 객체 메서드에 대한 정보 가지고 오기 ")
+    void showClass_3() {
+        Class<Car> carClass = Car.class;
+        Method[] methods = carClass.getDeclaredMethods();
+        Arrays.stream(methods).forEach(method -> {
+            logger.info("Method: " + method.getName() + ", Return Type: " + method.getReturnType());
+        });
+    }
+
+    @Test
+    @DisplayName("test로 시작하는 메서드 실행하기")
+    void testMethodRun() throws Exception {
+        Class<?> carClass = Car.class;
+        Object carInstance = carClass.getDeclaredConstructor().newInstance(); // Car 인스턴스 생성
+        Method[] methods = carClass.getDeclaredMethods(); // Car 클래스의 메서드들 가져오기
+
+        Arrays.stream(methods)
+                .filter(method -> method.getName().startsWith("test"))
+                .forEach(method -> {
+                    try {
+                        logger.info("invoke: " + method.invoke(carInstance));
+                        method.invoke(carInstance);
+                    } catch (IllegalAccessException | InvocationTargetException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+    }
+
+    @Test
+    @DisplayName("@PrintView 애노테이션 메서드 실행 ")
+    void annotationMethodRun() throws Exception {
+        Class<?> carClass = Car.class;
+        Object carInstance = carClass.getDeclaredConstructor().newInstance();
+        Method[] methods = carClass.getDeclaredMethods();
+        // 기존 System.out을 임시로 바꿀 스트림 생성
+
+        ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+        PrintStream originalOut = System.out;
+        System.setOut(new PrintStream(outContent)); // System.out을 ByteArrayOutputStream으로 설정
+
+        Arrays.stream(methods)
+                .filter(method -> method.isAnnotationPresent(PrintView.class))
+                .forEach(method -> {
+                    try {
+                        method.invoke(carInstance);
+                    } catch (IllegalAccessException | InvocationTargetException e) {
+                        throw new RuntimeException(e);
+                    }
+            });
+        // 메서드 호출 후 출력된 내용을 String으로 변환
+        String output = outContent.toString().trim();
+        logger.info("invoke: " + output);
+
+        // assertThat으로 출력된 내용을 검증
+        assertThat(output).isEqualTo("자동차 정보를 출력 합니다.");
+        }
```

Comment:
```suggestion
    }
```

코드 정렬이 제대로 안되어있네요 ㅜㅜ 
자동 포맷팅을 활용해보아요~